src/global_distrib/mod_global_distrib_utils.erl

```diff
@@ -46,7 +46,7 @@ binary_to_metric_atom(Binary) ->
     list_to_atom(List).
 
 ensure_metric(Metric, Type) ->
```

Comment:
should not be really here.
One of our clients (c) has this code in mongoose_metrics:

```erlang
-spec ensure_update(Host :: ejabberd:lserver() | global,                             
                    Name :: term() | list(),                                         
                    Change :: term(),                                                
                    Type :: atom()) -> any().                                        
ensure_update(Host, Name, Value, Type) when is_atom(Type) ->                         
    Res = mongoose_metrics:update(Host, Name, Value),                                
    case Res of                                                                      
        {error,not_found} ->                                                         
            mongoose_metrics:ensure_metric(Host, Name, Type),                        
            mongoose_metrics:update(Host, Name, Value);                              
        _ ->                                                                         
            Res                                                                      
    end.  

-spec ensure_metric(ejabberd:server() | global, metric(), atom() | tuple(), atom()) -> ok.
ensure_metric(Host, Metric, Type, ShortType) when is_atom(Metric) ->                 
    ensure_metric(Host, [Metric], Type, ShortType);                                  
ensure_metric(Host, Metric, Type, ShortType) when is_list(Metric) ->                 
    %% the split into ShortType and Type is needed because function metrics are      
    %% defined as tuples (that is Type), while exometer:info returns only 'function' 
    PrefixedMetric = name_by_all_metrics_are_global(Host, Metric),                   
    case exometer:info(PrefixedMetric, type) of                                      
        ShortType -> {ok, already_present};                                          
        undefined ->                                                                 
            Result =                                                                 
                try                                                                  
                    exometer:new(PrefixedMetric, Type)                               
                catch error:exists ->                                                
                          ok                                                         
                end,                                                                 
            %% exometer_report has a dead lock, so make it in a separate process     
            proc_lib:spawn(fun() ->                                                  
               init_metric_subscriptions(PrefixedMetric, Type)                       
               end),                                                                 
            Result                                                                   
    end. 
```

We can prepare a patch.

This TODO is not relevant to this PR.

Reply:
I'll create "Metrics refactoring" project.